tests/app/views/test_highlight.py

```diff
@@ -0,0 +1,112 @@
+
+import json
+
+from app.main.services import search_service
+
+from tests.helpers import (
+    BaseApplicationTestWithIndex,
+    make_search_api_url,
+    make_service,
+)
+
+
+class TestHighlightedService(BaseApplicationTestWithIndex):
+
+    def setup(self):
+        super().setup()
+
+        def put(s):
+            response = self.client.put(
+                make_search_api_url(s),
+                data=json.dumps(s),
+                content_type="application/json",
+            )
+            assert response.status_code == 200
+            return response
+
+        with self.app.app_context():
+            put(make_service(
+                id="1",
+                serviceDescription="Accessing, storing and retaining email.",
+                lot="cloud-support",
+            ))
+            put(make_service(
+                id="2",
+                serviceDescription="The <em>quick</em> brown fox jumped over the <em>lazy</em> dog.",
+                lot="mother-goose",
+            ))
+            put(make_service(
+                id="3",
+                serviceDescription=(
+                    # serviceDescription is limited to 500 characters by validator in
+                    # digitalmarketplace-frameworks/frameworks/g-cloud-10/questions/services
+                    "This service description has 500 characters. "
+                    "It is made of 5 repetitions of a 100 character string.\n"
+                    * 5
+                ),
+                lot="long-text",
+            ))
+
+            search_service.refresh("test-index")
+
+    def test_search_results_have_highlighted_service_description(self):
+        search_results = self.client.get("/test-index/services/search").json["documents"]
+        assert all(doc["highlight"]["serviceDescription"] for doc in search_results)
+
+    def test_highlighted_service_description_has_list_containing_service_description(self):
+        search_results = self.client.get("/test-index/services/search").json["documents"]
+        cloud_support = [doc for doc in search_results if doc["id"] == "1"][0]["highlight"]["serviceDescription"][0]
+        assert (
+            cloud_support
+            ==
+            "Accessing, storing and retaining email."
+        )
+        long_text = [doc for doc in search_results if doc["id"] == "3"][0]["highlight"]["serviceDescription"][0]
+        assert len(long_text) == 500
+
+        search_results = self.client.get("/test-index/services/search?q=long-text").json["documents"]
```

Comment:
Should we split the two GETs into two test scenarios?

Reply:
Good idea, I created a new parametrised test `test_highlighted_service_description_always_contains_full_service_description` so that the purpose of the two gets is clearer

Reply:
Fab - we could prob parametrize the last test as well?

Reply:
Unfortunately it can't be easily parametrized, as we're asserting different things each time; I can split it into two tests, or alternatively change the name to `test_html_in_highlighted_service_description_is_always_escaped`